app/src/main/java/com/github/gotifynotiftester/services/CustomNotif.kt

```diff
@@ -0,0 +1,102 @@
+package com.github.gotifynotiftester.services
+
+import android.app.Notification
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.Service
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+import android.os.IBinder
+import android.os.Looper
+import android.os.Messenger
+import com.github.gotify.connector.GotifyMessage
+import com.github.gotify.connector.GotifyServiceHandler
+import com.github.gotify.connector.getGotifyIdInSharedPref
+import java.util.concurrent.ThreadLocalRandom
+
+const val customServiceName = ".services.CustomNotif" // ! Starts with a dot
+
+class CustomNotif : Service(){
+
+    private var notifier: Notifier? = null
+
+    private val notifMessenger = Messenger(MessageHandler(this))
+
+    internal class MessageHandler(var service: CustomNotif) : GotifyServiceHandler(Looper.getMainLooper()){
+        override fun onMessage(message: GotifyMessage) {
+            val text = message.message
+            var title = message.title
```

Comment:
You can use the [elvis operator](https://kotlinlang.org/docs/reference/null-safety.html#elvis-operator) here
```suggestion
            val title = message.title ?: service.applicationInfo.name
```
I think gotify guarantees, that the title is set. If it is not set on POST /message, then the application name will be used.